### PR TITLE
Improve the caller/callee correlation diagnostics

### PR DESCRIPTION
**Pull request by kohsuke**
I was trying to diagnose a remoting call that bridges two VMs that is hanging. This can be made easier if thread dump allows us to correlate the caller/callee.

This change adds the request ID to thread name so that thread dump will show this clearly

Also fixes one incorrect use of `toString()` in diagnosis, which should be `getName()` --- see all the other uses of thread names.

@reviewbybees

### CODE REVIEW

**Comment by recena** [comment]
LGTM, 🐝

**Comment by stephenc** [comment]
:bee:

File: src/main/java/hudson/remoting/Request.java [hunk 1]
```diff
@@ -233,6 +233,8 @@ public boolean isDone() {
 
             public RSP get() throws InterruptedException, ExecutionException {
                 synchronized(Request.this) {
+                    String oldThreadName = Thread.currentThread().getName();
```
Comment (kohsuke) [inline src/main/java/hudson/remoting/Request.java]: This is the same variable name used everywhere. I think it's `old (thread name)` not `(old thread) name`

**Comment by oleg-nenashev** [comment]
Post-:bee:

**Comment by oleg-nenashev** [comment]
I will add the post-factum JIRA issue since this stabilization branch is a subject for careful reviews

**Comment by oleg-nenashev** [comment]
https://issues.jenkins-ci.org/browse/JENKINS-39543

**Pull request merged**